language/move-stdlib/sources/string.move

```diff
@@ -85,10 +85,37 @@ module std::string {
         internal_index_of(&s.bytes, &r.bytes)
     }
 
+    public fun next_char_boundary(s: &String, i: u64): u64 {
+        internal_next_char_boundary(&s.bytes, i)
+    }
+
+    /// Returns the total number of characters in a string
+    public fun chars_count(s: &String): u64 {
+        internal_chars_count(&s.bytes)
+    }
+
+    /// Returns a sub-string using the given character indices, where `i` is the start character index and `j` end character index.
+    public fun sub_string_char(s: &String, i: u64, j: u64): String {
+        String {
+            bytes: internal_sub_string_char(&s.bytes, i, j)
+        }
+    }
+
+
+    /// Insert the other string at a character index in given string.
+    public fun insert_char(s: &mut String, at: u64, c: String) {
+        let result = internal_insert_char(&s.bytes, at, &c.bytes);
+        s.bytes = result
+    }
```

Comment:
I wonder if we need/want these functions. Just a thought but I wonder if what you want instead is a `Char` type and then expose something like `to_chars` and `from_chars` that work over `vector<Char>`. If you want to do inserts around unicode scalar boundaries, that feels like the right place to do it. (And it is a bit more generalized). 

And if you really want the functions he added, you can compose `insert` and `next_char_boundary` (thought it will needlessly rerun utf validation?